src/array_api_stubs/_draft/array_object.py

```diff
@@ -52,9 +58,10 @@ def mT(self: array) -> array:
         out: array
             array whose last two dimensions (axes) are permuted in reverse order relative to original array (i.e., for an array instance having shape ``(..., M, N)``, the returned array must have shape ``(..., N, M)``). The returned array must have the same data type as the original array.
         """
+        ...
 
     @property
-    def ndim(self: array) -> int:
+    def ndim(self) -> int:
```

Comment:
So I see methods/properties which don't return arrays don't get any type hint for their `self` arg—could these be hinted with `Array`?

Reply:
Static type checkers don't like this because the ``TypeVar`` is not properly bound — see https://peps.python.org/pep-0484/#scoping-rules-for-type-variables ("Unbound type variables should not appear in the bodies of generic functions, or in the class bodies apart from method definitions:").
In general, there's no need to type hint the self arg unless the return type depends on the type of self — e.g. see https://peps.python.org/pep-0484/#user-defined-generic-types.

Reply:
A nice thing about using Protocol is that ``self`` is understood by static type checkers to be the ``Protocol``.

Reply:
Why is this only done for certain methods?

Reply:
I did it for any methods I looked at over the course of the PR. This is by no means comprehensive and should be done: here or elsewhere.